build-tools/packages/build-cli/src/commands/generate/layerGeneration.ts

```diff
@@ -172,12 +173,13 @@ export function maybeGetNewGeneration(
 	}
 
 	const previousPkgVersion = match[3];
-	if (previousPkgVersion === currentPkgVersion) {
-		log.verbose(
-			`Package version has not changed (still ${currentPkgVersion}); skipping generation update.`,
-		);
+	const result = diff(currentPkgVersion, previousPkgVersion);
+	// Only "minor" or "major" version changes trigger generation updates.
+	if (result === null || (result !== "minor" && result !== "major")) {
+		log.verbose(`No minor or major release since last update; skipping generation update.`);
```

Comment:
If `previousPkgVersion` is 3.0.0 and `currentPkgVersion`, is 3.1.1 then I think that would result in updating the generation. I imagine this could happen if 3.1.0 was released 30 days after 3.0.0 and then the patch release was a few days later. Maybe we make it so if `currentPkgVersion` is `X.Y.Z` and `Z` is not zero we don't increment (since it should always be 0 immediately after a major/minor bump).

Reply:
Fixed it. Also, added a test case for this scenario.